cli: check MVCCStats in `debug check-store`

tbg opened the pull request:
Unfortunately, this discovered #20180: The stats essentially never match (or this PR has a bug).

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/20181)
<!-- Reviewable:end -->

petermattis commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> 

I don't see anything obviously wrong with the code, but I'm very suspicious there is something. Don't we check on disk stats vs computed stats elsewhere (e.g. during splits)?

---

Review status: 0 of 1 files reviewed at latest revision, all discussions resolved, all commit checks successful.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/20181#-:-KzU3hEyaOVd6QY4WFbF:b39ynjl)*
<!-- Sent from Reviewable.io -->

tschottdorf commented:
@petermattis no. I hoped that this would be fixed by #21070 but it's still happening. I'm working on making the replica GC queue check this (which I need anyway to trigger repair now that the stats are at least somewhat fixed, and more fixes are planned).

tbg committed:
cli: check MVCCStats in `debug check-store`

Unfortunately, this discovered #20180: The stats essentially never match (or
this PR has a bug).

Release note: None
